src/db_adapters/access_keys.rs

```diff
@@ -33,11 +34,24 @@ pub(crate) async fn handle_access_keys(
         .filter_map(|outcome_with_receipt| outcome_with_receipt.receipt.as_ref());
 
     let mut access_keys = HashMap::<(String, String), models::access_keys::AccessKey>::new();
+    let mut deleted_accounts = HashMap::<String, String>::new();
 
     for receipt in successful_receipts {
         if let near_primitives::views::ReceiptEnumView::Action { actions, .. } = &receipt.receipt {
             for action in actions {
                 match action {
+                    near_primitives::views::ActionView::DeleteAccount { .. } => {
+                        deleted_accounts.insert(
+                            receipt.receiver_id.to_string(),
+                            receipt.receipt_id.to_string(),
+                        );
+                        access_keys.iter_mut().for_each(|(key, value)| {
+                            if key.1 == receipt.receiver_id.to_string() {
```

Comment:
I would expect this to work fine :thinking: 

```suggestion
                            if key.1 == receipt.receiver_id {
```